src/utils/helpers.ts

```diff
@@ -429,21 +429,32 @@ export const subIn256 = (x: bigint, y: bigint): bigint => {
   }
 };
 
-export const splitFee = (
+export const calculateFees = (
   amount: bigint,
-  settlerFeesInBps: bigint,
-  protocolShareBps: bigint
-): { amountIn: bigint; protocolShareAmount: bigint; senderShareAmount: bigint } => {
-  const feeAmount = (amount * settlerFeesInBps) / 10_000n;
-  const amountIn = amount - feeAmount;
+  senderShareBps: bigint,
+  protocolShareBps: bigint,
+  protocolShareOfSenderFeePct: bigint
+): { amountIn: bigint; protocolFee: bigint; senderFee: bigint } => {
+  if (protocolShareBps + senderShareBps > 200n) {
+    throw new Error('max fee exceeded: sum of protocolShareBps, senderShareBps > 200');
+  }
+
+  let protocolFee = (amount * protocolShareBps) / 10_000n;
+  let senderFee = (amount * senderShareBps) / 10_000n;
 
-  const protocolShareAmount = (feeAmount * protocolShareBps) / settlerFeesInBps;
-  const senderShareAmount = feeAmount - protocolShareAmount;
+  if (protocolShareOfSenderFeePct > 0n) {
+    const protocolFeeFromSenderFee = (senderFee * protocolShareOfSenderFeePct) / 10_000n;
```

Comment:
This should be divide by 100 (https://github.com/meliopolis/chainhopper-protocol/blob/main/src/base/Settler.sol#L28C31-L28C36).